### PR TITLE
Search display Tree in Result view

### PR DESCRIPTION
**Pull request by georgiana-ecobici**
Change-Id: I363b4e190d2b90ef4250bccd88346935b9773ee5
Signed-off-by: Georgiana Ecobici <georgiana-elena.ecobici@thalesgroup.com>

### CODE REVIEW

File: core/plugins/org.polarsys.capella.core.ui.search/src/org/polarsys/capella/core/ui/search/AbstractCapellaSearchEntry.java [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+   * Set to true if a real match (eg. not an intermediary structural object for ordering purpose only), false otherwise
+   */
+  public void setMatchedOnce(boolean matchedOnce) {
+    this.matchedOnce = matchedOnce;
+  }
+
+  /**
+   * @return AbstractCapellaSearchResultEntry hierarchy in a root to leaf order (eg. tree like), void Stack otherwise.
+   */
+  public Stack<AbstractCapellaSearchEntry> getHierarchyFromRootToLeaf() {
```
Comment (minhtutonthat) [inline core/plugins/org.polarsys.capella.core.ui.search/src/org/polarsys/capella/core/ui/search/AbstractCapellaSearchEntry.java]: Deque, the unsynchronized version of Stack, should be used to be more performance efficient.

File: core/plugins/org.polarsys.capella.core.ui.search/src/org/polarsys/capella/core/ui/search/CapellaSearchResult.java [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+      for (Object currentEntrySubHierarchy : currentEntrySubHierarchyCollection) {
+        if (alreadyExist = currentEntrySubHierarchy.equals(entryToInsert)) {
+          updateInsertionPoint((AbstractCapellaSearchEntry) currentEntrySubHierarchy, (AbstractCapellaSearchEntry) entryToInsert);
+          insert3(currentEntrySubHierarchy, entryToInsert.getChildren(), notify);
+          break;
+        }
+      }
+      if (!alreadyExist) {
+        currentEntrySubHierarchyCollection.add(entryToInsert);
+        if (notify) {
```
Comment (minhtutonthat) [inline core/plugins/org.polarsys.capella.core.ui.search/src/org/polarsys/capella/core/ui/search/CapellaSearchResult.java]: This has not been finished, isn't it?

File: core/plugins/org.polarsys.capella.core.ui.search/src/org/polarsys/capella/core/ui/search/CapellaSearchResult.java [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+          return occurence;
+        }
+      }
+    }
+    occurence = new CapellaSearchMatchOccurence(entryHierarchyIntoWhichInsert, eTypedElem, text,
+        true, (IProject)file);
+    
+    entryHierarchyIntoWhichInsert.addChildren(occurence);
+
+    if (notify) {
```
Comment (minhtutonthat) [inline core/plugins/org.polarsys.capella.core.ui.search/src/org/polarsys/capella/core/ui/search/CapellaSearchResult.java]: Same remark as above

File: core/plugins/org.polarsys.capella.core.ui.search/META-INF/MANIFEST.MF [hunk 4]
```diff
@@ -3,18 +3,18 @@ Bundle-ManifestVersion: 2
...
 Bundle-Localization: plugin
+Require-Bundle: org.eclipse.search,
+ org.eclipse.jface.text,
+ org.polarsys.capella.core.platform.sirius.ui.navigator,
+ org.polarsys.capella.common.ui.toolkit,
+ org.eclipse.sirius.common.ui,
+ org.polarsys.capella.core.sirius.ui,
+ org.eclipse.ui.workbench.texteditor
+Export-Package: org.polarsys.capella.core.ui.search
+Import-Package: org.polarsys.capella.common.ui.toolkit.viewers.data
```
[the remark this answers is not in the document]
Comment (georgiana-ecobici) [reply core/plugins/org.polarsys.capella.core.ui.search/META-INF/MANIFEST.MF]: No specific reason, will use require bundle instead.

File: core/plugins/org.polarsys.capella.core.ui.search/src/org/polarsys/capella/core/ui/search/AbstractCapellaSearchEntry.java [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+   * all AbstractCapellaSearchResultEntry children, potentially matches themselves or intermediary structural object(s)
+   * for ordering purposes only)
+   */
+  private Collection<Object> children;
+  
+  private String text;
+  
+  private IProject project;
+
+  public AbstractCapellaSearchEntry(AbstractCapellaSearchEntry _parent, Object _source, boolean _matched, IProject _project) {
```
[the remark this answers is not in the document]
Comment (georgiana-ecobici) [reply core/plugins/org.polarsys.capella.core.ui.search/src/org/polarsys/capella/core/ui/search/AbstractCapellaSearchEntry.java]: OK.

File: core/plugins/org.polarsys.capella.core.ui.search/src/org/polarsys/capella/core/ui/search/AbstractCapellaSearchEntry.java (hunk 1 above)
Comment (georgiana-ecobici) [reply core/plugins/org.polarsys.capella.core.ui.search/src/org/polarsys/capella/core/ui/search/AbstractCapellaSearchEntry.java]: OK.

File: core/plugins/org.polarsys.capella.core.ui.search/src/org/polarsys/capella/core/ui/search/CapellaSearchResult.java (hunk 2 above)
Comment (georgiana-ecobici) [reply core/plugins/org.polarsys.capella.core.ui.search/src/org/polarsys/capella/core/ui/search/CapellaSearchResult.java]: I didn't use the notify, will be removed.

File: core/plugins/org.polarsys.capella.core.ui.search/src/org/polarsys/capella/core/ui/search/CapellaSearchResult.java [hunk 6]
```diff
@@ -0,0 +1,300 @@
...
+   * @param currentEntrySubHierarchyCollection
+   * @param entryToInsert
+   * @param notify
+   */
+  private void insert2(Collection<Object> currentEntrySubHierarchyCollection, AbstractCapellaSearchEntry entryToInsert,
+      boolean notify) {
+
+      boolean alreadyExist = false;
+      for (Object currentEntrySubHierarchy : currentEntrySubHierarchyCollection) {
+        if (alreadyExist = currentEntrySubHierarchy.equals(entryToInsert)) {
```
[the remark this answers is not in the document]
Comment (georgiana-ecobici) [reply core/plugins/org.polarsys.capella.core.ui.search/src/org/polarsys/capella/core/ui/search/CapellaSearchResult.java]: Will remove this.

File: core/plugins/org.polarsys.capella.core.ui.search_cdb/src/org/polarsys/capella/core/ui/search_cdb/CapellaSearchResultTreeContentProvider.java [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+  public boolean hasChildren(Object element) {
+    if (element instanceof IProject) {
+      return true;
+    } else if (element instanceof CapellaSearchMatch_CDB) {
+      return !((CapellaSearchMatch_CDB) element).getMatchOccurrences().isEmpty();
+    }
+    return super.hasChildren(element);
+  }
+  
+  /*public Object[] getSources(Object[] input) {
```
Comment (Cosmin-Visan) [inline core/plugins/org.polarsys.capella.core.ui.search_cdb/src/org/polarsys/capella/core/ui/search_cdb/CapellaSearchResultTreeContentProvider.java]: Will this commented code be used in the future ?

Comment (georgiana-ecobici) [reply core/plugins/org.polarsys.capella.core.ui.search_cdb/src/org/polarsys/capella/core/ui/search_cdb/CapellaSearchResultTreeContentProvider.java]: No, I accidentally commited this file. I will remove it.

File: core/plugins/org.polarsys.capella.core.ui.search/src/org/polarsys/capella/core/ui/search/searchfor/ModelSearchImagesUtil.java [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+import org.eclipse.core.runtime.FileLocator;
+import org.eclipse.core.runtime.Path;
+import org.eclipse.jface.resource.ImageDescriptor;
+import org.eclipse.swt.graphics.Image;
+import org.osgi.framework.Bundle;
+
+/**
+ * Utility class for images
+ */
+public final class ModelSearchImagesUtil {
```
Comment (minhtutonthat) [inline core/plugins/org.polarsys.capella.core.ui.search/src/org/polarsys/capella/core/ui/search/searchfor/ModelSearchImagesUtil.java]: There are some problems with this class. It not only recreates images but also maintains a map of ImageDescriptor which is already done with the ExtendedImageRegistry. This causes performance issue and code redundancy. Moreover, it's not generic. Capella's elements are treated separately than diagram elements, viewpoint elements, etc.
IMO, there are 2 solutions: 1) Create dummy objects of eClasses just to get their label and image (using EObjectLabelProviderHelper). We've already done this in Capella in org.polarsys.capella.core.ui.metric.actions.MetricAction.
2) More complicated but I think feasible. Contribute a new class (e.g. EClassLabelProviderHelper) to get label and image from EClass. From an EClass we can get the Package URI and then get the ItemProviderAdapterFactory (contributed via org.eclipse.emf.edit.itemProviderAdapterFactories extension point). From the ItemProviderAdapterFactory we can determine the .edit bundle (e.g. org.polarsys.capella.core.data.gen.edit in Capella). With the bundle we can get the image (like what you've done) with ExtendedImageRegistry.INSTANCE.getImageDescriptor. For the label, we can inspire from the method org.eclipse.emf.common.EMFPlugin.InternalHelper.getString(String, boolean).

File: core/plugins/org.polarsys.capella.core.ui.search/src/org/polarsys/capella/core/ui/search/CapellaSearchPage.java [hunk 9]
```diff
@@ -0,0 +1,389 @@
...
+import org.eclipse.swt.widgets.Label;
+import org.eclipse.swt.widgets.Table;
+import org.eclipse.ui.dialogs.PatternFilter;
+import org.eclipse.ui.fieldassist.ContentAssistCommandAdapter;
+import org.eclipse.ui.texteditor.ITextEditorActionDefinitionIds;
+import org.polarsys.capella.core.model.handler.command.CapellaResourceHelper;
+import org.polarsys.capella.core.sirius.ui.helper.SessionHelper;
+import org.polarsys.capella.core.ui.search.searchfor.CapellaLeftSearchForContainerArea;
+import org.polarsys.capella.core.ui.search.searchfor.CapellaRightSearchForContainerArea;
+import org.polarsys.capella.test.framework.helpers.TestHelper;
```
[the remark this answers is not in the document]
Comment (georgiana-ecobici) [reply core/plugins/org.polarsys.capella.core.ui.search/src/org/polarsys/capella/core/ui/search/CapellaSearchPage.java]: Ok

File: core/plugins/org.polarsys.capella.core.ui.search/src/org/polarsys/capella/core/ui/search/CapellaSearchPage.java [hunk 10]
```diff
@@ -0,0 +1,389 @@
...
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *  
+ * Contributors:
+ *    Thales - initial API and implementation
+ *******************************************************************************/
+package org.polarsys.capella.core.ui.search;
+
+import java.util.ArrayList;
```
[the remark this answers is not in the document]
Comment (georgiana-ecobici) [reply core/plugins/org.polarsys.capella.core.ui.search/src/org/polarsys/capella/core/ui/search/CapellaSearchPage.java]: OK

File: core/plugins/org.polarsys.capella.core.ui.search/src/org/polarsys/capella/core/ui/search/CapellaSearchSettingsHistory.java [hunk 11]
```diff
@@ -0,0 +1,164 @@
...
+ *    Thales - initial API and implementation
+ *******************************************************************************/
+package org.polarsys.capella.core.ui.search;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.jface.dialogs.IDialogSettings;
+
+public class CapellaSearchSettingsHistory {
```
[the remark this answers is not in the document]
Comment (georgiana-ecobici) [reply core/plugins/org.polarsys.capella.core.ui.search/src/org/polarsys/capella/core/ui/search/CapellaSearchSettingsHistory.java]: Ok

File: core/plugins/org.polarsys.capella.core.ui.search/src/org/polarsys/capella/core/ui/search/CapellaSearchPage.java (hunk 9 above)
[the remark this answers is not in the document]
Comment (georgiana-ecobici) [reply core/plugins/org.polarsys.capella.core.ui.search/src/org/polarsys/capella/core/ui/search/CapellaSearchPage.java]: Done.

File: core/plugins/org.polarsys.capella.core.ui.search/src/org/polarsys/capella/core/ui/search/searchfor/MetaClassesParticipantsItemProvider.java [hunk 12]
```diff
@@ -0,0 +1,165 @@
...
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.ResourcesPlugin;
+import org.eclipse.emf.common.util.EList;
+import org.eclipse.emf.ecore.EAttribute;
+import org.eclipse.emf.ecore.EClass;
+import org.eclipse.emf.ecore.EClassifier;
+import org.eclipse.emf.ecore.EPackage;
+import org.eclipse.sirius.business.api.dialect.DialectManager;
+import org.eclipse.sirius.business.api.session.Session;
+import org.eclipse.sirius.diagram.DiagramPackage;
```
[the remark this answers is not in the document]
Comment (georgiana-ecobici) [reply core/plugins/org.polarsys.capella.core.ui.search/src/org/polarsys/capella/core/ui/search/searchfor/MetaClassesParticipantsItemProvider.java]: I have: org.eclipse.sirius.common.ui.

File: core/plugins/org.polarsys.capella.core.ui.search/src/org/polarsys/capella/core/ui/search/CapellaSearchPage.java (hunk 10 above)
[the remark this answers is not in the document]
Comment (georgiana-ecobici) [reply core/plugins/org.polarsys.capella.core.ui.search/src/org/polarsys/capella/core/ui/search/CapellaSearchPage.java]: Done.

File: core/plugins/org.polarsys.capella.core.ui.search/src/org/polarsys/capella/core/ui/search/searchfor/AbstractCapellaSearchForContainerArea.java [hunk 13]
```diff
@@ -0,0 +1,282 @@
...
+import org.eclipse.swt.widgets.Group;
+import org.eclipse.ui.IWorkbench;
+import org.eclipse.ui.PlatformUI;
+import org.eclipse.ui.dialogs.PatternFilter;
+import org.eclipse.ui.progress.IProgressService;
+import org.polarsys.capella.core.ui.search.Activator;
+import org.polarsys.capella.core.ui.search.Messages;
+import org.polarsys.capella.core.data.capellamodeller.provider.CapellaModellerEditPlugin;
+
+public abstract class AbstractCapellaSearchForContainerArea {
```
[the remark this answers is not in the document]
Comment (georgiana-ecobici) [reply core/plugins/org.polarsys.capella.core.ui.search/src/org/polarsys/capella/core/ui/search/searchfor/AbstractCapellaSearchForContainerArea.java]: II removed deprecated except two more which I noted it: PatternConstructor.createPattern, viewer.setAllChecked.

File: core/plugins/org.polarsys.capella.core.ui.search/src/org/polarsys/capella/core/ui/search/searchfor/CapellaLeftSearchForContainerArea.java [hunk 14]
```diff
@@ -0,0 +1,273 @@
...
+    try {
+      service.run(false, false, runnable);
+    } catch (InterruptedException e) {
+      e.printStackTrace();
+    } catch (InvocationTargetException e) {
+      e.printStackTrace();
+    }
+  }
+
+  public void addFilters() {
```
Comment (georgiana-ecobici) [inline core/plugins/org.polarsys.capella.core.ui.search/src/org/polarsys/capella/core/ui/search/searchfor/CapellaLeftSearchForContainerArea.java]: I will delete this function in next commit.

**Pull request merged**